docs/source/cc_service.md

```diff
@@ -0,0 +1,297 @@
+if [ -f $BLD/connection.json ]; then
+   mkdir -p "$RELEASE"/chaincode/server
+       cp $BLD/connection.json "$RELEASE"/chaincode/server
+       exit 0
+fi
+
+exit 1
+```    
+
+### /bin/run
+
+This script is not required but is included for completeness. This sample assumes that the chaincode runtime is managed independently and simply returns success.
+
+```sh
+#!/bin/bash
+
+  set -euo pipefail
+
+  # if the runtime is managed externally, we don’t have to start the chaincode from this script
+  # just return success
+
+  exit 0
+```
+
+## Writing chaincode to run as an external service
+
+Currently, the chaincode as an external service model model is only supported by GO chaincode shim. In Fabric v2.0, the GO shim API adds a `ChaincodeServer` type that developers should use to create a chaincode server.  The `Invoke` and `Query` APIs are unaffected. Developers should write to the `shim.ChaincodeServer` API, then build the chaincode and run it in the external environment of choice. Here is a simple sample chaincode program to illustrate the pattern:
+
+```go
+
+package main
+
+import (
+        "fmt"
+
+        "github.com/hyperledger/fabric-chaincode-go/shim"
+        pb "github.com/hyperledger/fabric-protos-go/peer"
+)
+
+// SimpleChaincode example simple Chaincode implementation
+type SimpleChaincode struct {
+}
+
+func (s *SimpleChaincode) Init(stub shim.ChaincodeStubInterface) pb.Response {
+        // init code
+}
+
+func (s *SimpleChaincode) Invoke(stub shim.ChaincodeStubInterface) pb.Response {
+        // invoke code
+}
+
+//NOTE - parameters such as ccid and endpoint information are hard coded here for illustration. This can be passed in in a variety of standard ways
+func main() {
+       //The ccid is assigned to the chaincode on install (using the “peer lifecycle chaincode install <package>” command) for instance
+        ccid := "mycc:fcbf8724572d42e859a7dd9a7cd8e2efb84058292017df6e3d89178b64e6c831"
+
+        server := &shim.ChaincodeServer{
+                        CCID: ccid,
+                        Address: "myhost:9999"
+                        CC: new(SimpleChaincode),
+                        TLSProps: shim.TLSProperties{
+                                Disabled: true,
+                        },
+                }
+        err := server.Start()
+        if err != nil {
+                fmt.Printf("Error starting Simple chaincode: %s", err)
+        }
+}
+```
+The key to running the chaincode as an external service is the use of `shim.ChaincodeServer`. This uses the new shim API `shim.ChaincodeServer` with the chaincode service properties described below:
+
+* **CCID** (string)- CCID should match chaincode's package name on peer. This the `CCID` associated with the installed chaincode as returned by the `peer lifecycle chaincode install <package>` CLI command. This can be obtained post-install using the "peer lifecycle chaincode queryinstalled" command.
```

Comment:
```suggestion
* **CCID** (string)- CCID should match chaincode's package name on peer. This is the `CCID` associated with the installed chaincode as returned by the `peer lifecycle chaincode install <package>` CLI command. This can be obtained post-install using the "peer lifecycle chaincode queryinstalled" command.
```